### PR TITLE
Treat .js as JSX

### PR DESCRIPTION
**Pull request by RyanCavanaugh**

### CODE REVIEW

**Comment by mhegazy** [comment]
We should add a unit test with --allowJs and some JSX syntax in there, and make sure no errors are reported.

**Comment by mhegazy** [comment]
:+1:

**Comment by DanielRosenwasser** [comment]
Why exactly are we doing this?

**Comment by billti** [comment]
@DanielRosenwasser Because ReactNative uses JSX directly in .js files, so this is becoming popular. VSCode has asked for this. As JSX is a superset, there is no ambiguity with plain JavaScript.

**Comment by billti** [comment]
Need to update the tests also. As expected, some errors changed now that parsing default has changed (though only 2, and the errors look like an improvement - good job! :-) ).

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -546,7 +546,7 @@ namespace ts {
 
         function getLanguageVariant(fileName: string) {
             // .tsx and .jsx files are treated as jsx language variant.
-            return fileExtensionIs(fileName, ".tsx") || fileExtensionIs(fileName, ".jsx") ?  LanguageVariant.JSX  : LanguageVariant.Standard;
+            return fileExtensionIs(fileName, ".tsx") || fileExtensionIs(fileName, ".jsx") || fileExtensionIs(fileName, '.js') ?  LanguageVariant.JSX  : LanguageVariant.Standard;
```
Comment (DanielRosenwasser) [inline src/compiler/parser.ts]: Double quotes.

**Pull request merged**